src/main/resources/com/google/api/codegen/go/doc.snip

```diff
@@ -10,6 +10,15 @@
     @join line : view.packageDoc
         // {@line}
     @end
+    //
+    // Use of Context
+    //
+    // The ctx passed to NewClient is used for authentication requests and gRPC
+    // transport setup, but not for future calls performed using the client.
+    // These calls use the ctx passed to individual methods on the client.
+    //
+    // Using deadlines on requests is recommended. See more information at
```

Comment:
Another common pattern is to use the `r.Context()` from HTTP handlers.

The context is canceled if the client goes away, and so with it any in-flight gRPC requests.

I think I'd prefer us link off to some other more detailed docs than provide a one-liner that might be taken to extremes (e.g., setting deadlines on every request, rather than a deadline for a series of requests).

Reply:
> Another common pattern is to use the r.Context() from HTTP handlers.

I'm not grokking this. People use req.Context() as the input to our requests?

> The context is canceled if the client goes away, and so with it any in-flight gRPC requests.

What client? The http client? What does "go away" mean?

Reply:
> I think I'd prefer us link off to some other more detailed docs than provide a one-liner that might be taken to extremes (e.g., setting deadlines on every request, rather than a deadline for a series of requests).

This makes sense to me. It's definitely a nuanced topic.

Reply:
> I'm not grokking this. People use req.Context() as the input to our requests?

Yeah!

```
func handler(w http.ResponseWriter, r *http.Request) {
   ctx := r.Context()

   datastoreClient.Get(ctx, ...)
}
```

> What client? The http client? What does "go away" mean?

yep. e.g., if someone closed the tab or navigated away.

https://golang.org/pkg/net/http/#Request.Context

"For incoming server requests, the context is canceled when the client's connection closes, the request is canceled (with HTTP/2), or when the ServeHTTP method returns."

Reply:
Gotcha. Thanks! I don't think any of this precludes advice telling people to use deadlines. Folks should ideally do,

```
func handler(w http.ResponseWriter, r *http.Request) {
   ctx := context.WithDeadline(r.Context(), <something reasonable>)

   datastoreClient.Get(ctx, ...)
}
```

Apart from generally being good to set reasonable limits on requests, it's also good because of our retry policies (which have no timeout).

Reply:
Can we take the more detailed advice/scenarios and include them in the repo-level documentation, then link back to that?

`For information about setting deadlines, reusing contexts, and more please visit godoc.org/cloud.google.com/go#hdr-Context`

Reply:
SGTM!

Reply:
Actually, it would be better to link to documentation that is more general, and not cloud specific. This generator supports other Google APIs, so hard coding Cloud doc links into the comments might be undesirable.

Reply:
Any suggestions on good sources to link to on this topic or should we be writing something ourselves? Might make for a good blog post...